WooCommerce/Classes/ViewRelated/Products/Variations/ProductVariationsViewModel.swift

```diff
@@ -48,4 +48,10 @@ extension ProductVariationsViewModel {
     func shouldShowMoreButton(for product: Product) -> Bool {
         product.attributesForVariations.isNotEmpty
     }
+
+    /// Defines if empty state screen should show guide for creating attributes
+    ///
+    func shouldShowCreatingAttributesGuide(for product: Product) -> Bool {
+        product.variations.isEmpty && product.attributesForVariations.isEmpty
+    }
```

Comment:
I really like to have this kind of logic in the view model. Especially because it allows us to test it easily.

Could you add some unit tests to it? :-)

Reply:
Absolutely!

Reply:
Hey @Ecarrion - I was writing the tests and found a case I have not covered: when the product has variation but no attribute (i.e all the attributes are removed). How should we display the empty state in this case? I'm currently not showing attribute guide:

<img src="https://user-images.githubusercontent.com/5533851/116431754-41e1fe80-a872-11eb-9774-207cb03e993e.png" width=300 />

Reply:
Thanks for confirming, the test has been updated 🙂